File: sandbox/matplotlib/simple_plot.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-

# see https://matplotlib.org/tutorials/index.html

import matplotlib.pyplot as plt
import numpy as np

t = np.arange(0.0, 2.0, 0.01)
s = np.sin(2*np.pi*t)
c = np.cos(2*np.pi*t)
plt.plot(t, s, label='sin')
plt.plot(t, c, label='cos')

plt.xlabel('time (s)')
plt.ylabel('voltage (mV)')
plt.title('About as simple as it gets, folks')
plt.grid(True)
plt.legend()
#plt.savefig("test.png")
plt.show()

File: sandbox/matplotlib/simple_plot.jl
# basic figure with julia

@time using Plots  # 6.9s
@time pyplot()  # 5.2s

t = collect(0.0:0.01:2.0)
s = sin.(2*pi*t)
c = cos.(2*pi*t)

@time f = plot(t, s, label="sin") # 11s
plot!(t, c, label="cos")
plot!(xlab="time (s)", ylab="voltage (mV)")
plot!(legend=:bottomleft)

@time savefig("test.png") # 5.8s

display(f)

println("done. press <ENTER> to quit")
readline()
